src/main/kotlin/fi/metatavu/timebank/api/severa/SeveraBearerTokenContainer.kt

```diff
@@ -0,0 +1,71 @@
+package fi.metatavu.timebank.api.severa
+
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import fi.metatavu.timebank.api.severa.models.SeveraAccessToken
+import okhttp3.FormBody
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.eclipse.microprofile.config.inject.ConfigProperty
+import org.slf4j.Logger
+import javax.enterprise.context.ApplicationScoped
+import javax.enterprise.context.RequestScoped
+import javax.inject.Inject
+
+@RequestScoped
+class SeveraBearerTokenContainer {
+    @ConfigProperty(name = "severa.base.url")
+    lateinit var severaBaseUrl: String
+
+    @ConfigProperty(name = "severa.client.id")
+    lateinit var severaClientId: String
+
+    @ConfigProperty(name = "severa.client.secret")
+    lateinit var severaClientSecret: String
+
+    @Inject
+    lateinit var logger: Logger
+
+    /**
+     * Generates a new access token for Severa API calls
```

Comment:
This doesn't generate an access token but rather asks Severa to generate one for us. Imo docs could just say `Gets a fresh access token for Severa API calls....`.

You'd also want this method to be private and expose another method from this class that's responsible for determining whether to use a potentially existing access token or to get a fresh one. Hence the word `container` in the classes name.

Reply:
Also IMO as we discussed yesterday, you could hard code the pre-defined scopes that you're going to need.